src/clients/c++/library/request_common.cc

```diff
@@ -218,8 +218,9 @@ InputImpl::SetSharedMemory(
         "The input '" + Name() + "' can only be set once with SetSharedMemory");
   }
 
-  // verify byte size of shared memory equals that of expected batch byte size
-  if (byte_size != (batch_size_ * byte_size_)) {
+  // Verify byte size of shared memory equals that of expected batch byte size
+  // Skip check for string and variable size tensors
+  if ((byte_size_ != -1) && (byte_size > (batch_size_ * byte_size_))) {
```

Comment:
This is for input so we should be able to have exact size testing. So shouldn't the ">" here be "!=" ?

Reply:
Yes, fixed it in the other batcher fix branch, reverted it back here as well.